packages/x-date-pickers-pro/src/MultiInputDateRangeField/MultiInputDateRangeField.tsx

```diff
@@ -13,17 +35,100 @@ export const MultiInputDateRangeField = React.forwardRef(function MultiInputDate
   TInputDate,
   TDate = TInputDate,
 >(inProps: MultiInputDateRangeFieldProps<TInputDate, TDate>, ref: React.Ref<HTMLInputElement>) {
-  const { startDate, endDate } = useMultiInputDateRangeField<
-    TInputDate,
-    TDate,
-    MultiInputDateRangeFieldProps<TInputDate, TDate>
-  >(inProps);
+  const themeProps = useThemeProps({
+    props: inProps,
+    name: 'MuiMultiInputDateRangeField',
+  });
+
+  const {
+    components,
+    componentsProps,
+    value,
+    defaultValue,
+    format,
+    onChange,
+    readOnly,
+    onError,
+    shouldDisableDate,
+    minDate,
+    maxDate,
+    disableFuture,
+    disablePast,
+    ...other
+  } = themeProps;
+
+  const ownerState = themeProps;
+
+  const Root = components?.Root ?? MultiInputDateRangeFieldRoot;
+  const rootProps = useSlotProps({
+    elementType: Root,
+    externalSlotProps: componentsProps?.root,
+    externalForwardedProps: other,
+    additionalProps: {
+      ref,
+    },
+    ownerState,
+  });
+
+  const Input = components?.Input ?? TextField;
+  const startInputProps = useSlotProps({
+    elementType: Input,
+    externalSlotProps: componentsProps?.input,
+    ownerState: { ...ownerState, position: 'start' },
+  });
+  const endInputProps = useSlotProps({
+    elementType: Input,
+    externalSlotProps: componentsProps?.input,
+    ownerState: { ...ownerState, position: 'end' },
+  });
```

Comment:
If I understand correctly, overriding `placeholder` could be done as follow:

```js
componentsProps: {
  input: ({ position }) => ({ placeholder: position==='start' ? 'Depature' : 'Arrival' })
}

Reply:
The placeholder is probably not a good example since the field is never empty.
The label would probably still be provided with `{ localText: { start: 'Departure', end: 'Arrival' } }` but that could be discussed.